pkgs/development/compilers/sbcl/common.nix

```diff
@@ -22,11 +22,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [texinfo];
 
-  patchPhase = ''
-    echo '"${version}.nixos"' > version.lisp-expr
```

Comment:
Maybe the `.nixos` suffix for the version was intentional? The [sbcl build system uses this file](https://github.com/sbcl/sbcl/blob/545d04f6ac15cdf678f97015c11efd9f92d8afee/generate-version.sh#L14-L26).

Reply:
I see what you mean.
I was about to revert that part, but it didn't seem to cause a problem with the build, so I checked how it's used.
In (doc/GIT-WORKFLOW.MD)[https://github.com/sbcl/sbcl/blob/master/doc/GIT-WORKFLOW.md] it says that this ceased to be useful after switching from CVS to a decentralised, branching VCS.
Thus, "in Git workflow, version.lisp-expr no longer exists in the repository, nor is the version number prepended to commit messages." 
There's more explanation in that file, but the upshot is that its use is deprecated if you're building from `git`.
`doc/GIT-WORKFLOW.md` is about 10 years old, and `sbc/default.nix` dates back 11 years to 2009, so I think this is a historical artifact that's survived until now simply because it didn't break anything.

Reply:
This is what it is set to in the relase tarball:

```
tar -xjOf /nix/store/k6gxslva6zvdgl0wbmgpgnh4qv55h3p3-sbcl-2.0.8-source.tar.bz2 sbcl-2.0.8/version.lisp-expr
;;; This file is auto-generated using generate-version.sh. Every time
;;; you re-run make.sh, this file will be overwritten if you are
;;; working from a Git checkout.
"2.0.8"
```

So it still holds that we overwrite this to say `2.0.8.nixos` which means we change something about the build process, rather than just removing unused code.

Reply:
It's not a hill I'm eager to die on, so I've reinstated that line.

Reply:
I guess an opinion from @7c6f434c would be interesting, as they are mantaining `sbcl`.

Reply:
I think we change some low-level system stuff (like uname call) so it makes sense to have a suffix, and we do not pass git to SBCL build, so it is not clear we should drop the version (and start-up banner does end up displaying this version, so it is not completely moot)